### PR TITLE
Add build Instructions for Windows environment with WSL

### PR DESCRIPTION
**Pull request by InfiniteEchoes**

### CODE REVIEW

**Review by clarus — APPROVED** [review]

File: docs/BUILD.md [hunk 1]
```diff
@@ -105,4 +106,38 @@ opam install --deps-only .
...
+   2. With `Ctrl+,` for VSCode's settings, checkout the environment 
+   settings for `Remote: [WSL ...]`. Modify the `Coq: Coq Project Root` 
+   to `.` , in particular, your preference folder with `_CoqProject` 
+   inside
+   3. Now you can `make` your project in WSL(not Windows) or customize 
+   your experience with other features of `CoqOfRust`
+
+### Known Issues
+
+WSL has a different format from Windows. Since we run Coq on WSL
```
Comment (olympichek) [inline docs/BUILD.md:135]: This "different format" discussion is somewhat confusing. What does it mean exactly?

File: docs/BUILD.md [hunk 2]
```diff
@@ -105,4 +106,38 @@ opam install --deps-only .
...
+only, the files being generated should be in WSL's file format. We 
+usually put the project on Windows, `make` it in WSL, to generate 
+Coq files of WSL's file format. The format differences here usually
+lead to significantly longer `make` time for WSL than other systems.
```
Comment (olympichek) [inline docs/BUILD.md:139]: Are there any downsides from putting project files within WSL, e.g. under the WSL `/home` directory?

File: docs/BUILD.md (hunk 1 above)
Comment (InfiniteEchoes) [reply docs/BUILD.md:135]: > This "different format" discussion is somewhat confusing. What does it mean exactly?

I admit that I'm doing lazy work here...I think I'll look into more details on this

File: docs/BUILD.md (hunk 2 above)
Comment (InfiniteEchoes) [reply docs/BUILD.md:139]: > Are there any downsides from putting project files within WSL, e.g. under the WSL `/home` directory?

I suppose projects under `/home` boosts the performance, as it's counted towards the WSL side. By default those projects should be accessed via `/mnt` for Windows.

I have actually experimented just putting the projects under `~` which is also `/home/my-user-name` before. It boosts.

File: docs/BUILD.md [hunk 3]
```diff
@@ -105,4 +106,38 @@ opam install --deps-only .
...
+
+### Known Issues
+
+WSL has a different format from Windows. Since we run Coq on WSL
+only, the files being generated should be in WSL's file format. We 
+usually put the project on Windows, `make` it in WSL, to generate 
+Coq files of WSL's file format. The format differences here usually
+lead to significantly longer `make` time for WSL than other systems. 
+
+As an alternative, you might copy the project under WSL's `/mnt` and
```
Comment (InfiniteEchoes) [inline docs/BUILD.md:141]: This should be considered fake. I'm making a new PR to fix all the places where I can fix.

**Pull request merged**